controlplane/kubeadm/internal/controllers/controller.go

```
@@ -145,19 +145,21 @@ func (r *KubeadmControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		return ctrl.Result{Requeue: true}, nil
+		return ctrl.Result{}, err
```

```suggestion
		return ctrl.Result{}, nil
```
Reconcile might be triggered by old event in the queue, but the object might not exist anymore when reconcile finally happen (and usually we ignore the error in those cases)

---

If the object has already beed deleted, the controller can catch in line:145. It will reach this line other than when the object was deleted.

---

Agree, l.145 will catch that case